### PR TITLE
[docs] Make Contributions/Building pages more prominent

### PR DESCRIPTION
**Pull request by richardliaw**
### index.rst
Screen Shot 2020-06-24 at 5 54 01 PM

### contributing page: 
Screen Shot 2020-06-24 at 5 49 16 PM

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/util/sgd/tests/test_tensorflow.py [hunk 1]
```diff
@@ -87,17 +87,17 @@ def test_save_and_restore(ray_start_2_cpus, num_replicas):  # noqa: F811
 
     shutil.rmtree(tmpdir)
 
-    model1_config = model1.get_config()
-    model2_config = model2.get_config()
-    assert _compare(model1_config, model2_config, skip_keys=["name"])
+    model1.get_config()
+    model2.get_config()
+    # assert _compare(model1_config, model2_config, skip_keys=["name"])
```
Comment (rkooo567) [inline python/ray/util/sgd/tests/test_tensorflow.py:92]: Maybe you should merge the latest master?

Comment (richardliaw) [reply python/ray/util/sgd/tests/test_tensorflow.py]: done

File: doc/source/debugging.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
+Debugging
+=========
```
Comment (richardliaw) [inline doc/source/debugging.rst:2]: this is just moved from `development.rst`

File: doc/source/index.rst [hunk 3]
```diff
@@ -137,6 +134,8 @@ Getting Involved
 
    ray-overview/index.rst
    installation.rst
+   development.rst
```
Comment (anabranch) [inline doc/source/index.rst:137]: I don't think this should be at the top level, can we create a new section that is "Contributing" below "Other Libraries" that is...


.. toctree::
   development.rst
   getting-involved.rst

I realize we want to encourage contributions. However, we want to focus on **users** of Ray, not just those that contribute and therefore that initial page real estate is really important.

File: doc/source/index.rst [hunk 4]
```diff
@@ -207,9 +206,8 @@ Getting Involved
 
 .. toctree::
    :maxdepth: -1
-   :caption: Development and Internals
+   :caption: Debugging and Internals
```
Comment (anabranch) [inline doc/source/index.rst:209]: We can keep this additional, although I would like to make a folder of 

"Debugging and Ray Internal"
internals/index.rst
internals/debugging.rst
internals/profiling.rst
internals/architecture.rst

Comment (anabranch) [reply doc/source/index.rst:209]: For the most part, it's a code smell to have everything at the top level folder like it would be with code

**Review by bllchmbrs — CHANGES_REQUESTED** [review]
Content: LGTM
Structure: would like another iteration please. As a general principle, don't do top level files unless completely necessary. Developing Ray << Using Ray in terms of priority.

overall a great start though and thank you for doing this!

File: doc/source/index.rst (hunk 4 above)
Comment (richardliaw) [reply doc/source/index.rst]: OK, this can be a later PR

**Comment by richardliaw** [comment]
@anabranch thanks for the feedback! appreciate your time. I've done another pass on the structure.

**Review by bllchmbrs — APPROVED** [review]
LGTM re:structure, seems like there's not much additional content. I might have a follow up PR for a slight reorganization but this one is good to go from my perspective.

**Comment by robertnishihara** [comment]
Looks good to me.

Not sure if it's related to this PR, but I see the following warnings when running `make html`.

```
debugging.rst:84: WARNING: duplicate label backend-logging, other instance in development.rst
ray/rllib/env/unity3d_env.py:docstring of ray.rllib.env.Unity3DEnv.step:5: WARNING: Unexpected indentation.
ray/rllib/env/unity3d_env.py:docstring of ray.rllib.env.Unity3DEnv.step:7: WARNING: Unexpected indentation.
ray/python/ray/rllib/env/unity3d_env.py:docstring of ray.rllib.env.Unity3DEnv.step:15: WARNING: Definition list ends without a blank line; unexpected unindent.
ray/python/ray/rllib/env/unity3d_env.py:docstring of ray.rllib.env.Unity3DEnv.step:17: WARNING: Unexpected indentation.
ray/python/ray/rllib/env/unity3d_env.py:docstring of ray.rllib.env.Unity3DEnv.step:18: WARNING: Block quote ends without a blank line; unexpected unindent.
```

File: doc/source/development.rst [hunk 5]
```diff
@@ -1,83 +1,30 @@
...
-Compilation
------------
-
-To speed up compilation, be sure to install Ray with the following commands:
-
-.. code-block:: shell
-
- cd python
- pip install -e . --verbose
+Buiilding Ray from Source
```
Comment (edoakes) [inline doc/source/development.rst:3]:
```suggestion
Building Ray from Source
```

**Review by edoakes — APPROVED** [review]
lgtm

**Pull request merged**